pkg/commands/artifact/run.go

```diff
@@ -434,18 +434,21 @@ func disabledAnalyzers(opts flag.Options) []analyzer.Type {
 		analyzers = append(analyzers, analyzer.TypeSecret)
 	}
 
-	// Filter only enabled misconfiguration scanners
-	ma, err := filterMisconfigAnalyzers(opts.MisconfigScanners, analyzer.TypeConfigFiles)
-	if err != nil {
-		log.Error("Invalid misconfiguration scanners specified, defaulting to use all misconfig scanners",
-			log.Any("scanners", opts.MisconfigScanners))
-	} else {
-		analyzers = append(analyzers, ma...)
-	}
-
 	// Do not perform misconfiguration scanning when it is not specified.
 	if !opts.Scanners.AnyEnabled(types.MisconfigScanner, types.RBACScanner) {
 		analyzers = append(analyzers, analyzer.TypeConfigFiles...)
+	} else {
+		// Filter only enabled misconfiguration scanners
+		ma, err := disabledMisconfigAnalyzers(analyzer.TypeConfigFiles, opts.MisconfigScanners)
+		if err != nil {
+			log.Error("Invalid misconfiguration scanners specified, defaulting to use all misconfig scanners",
+				log.Any("scanners", opts.MisconfigScanners))
+		} else {
+			analyzers = append(analyzers, ma...)
+		}
+
+		log.Debug("Enabling misconfiguration scanners",
+			log.Any("scanners", lo.Without(analyzer.TypeConfigFiles, analyzers...)))
```

Comment:
IIUC we don't need to check all analyzers here.
```suggestion
			log.Any("scanners", lo.Without(analyzer.TypeConfigFiles, ma...)))
```